src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -900,6 +888,33 @@ abstract class BTypes {
 
   final case class MethodBType(argumentTypes: List[BType], returnType: BType) extends BType
 
+  object BTypeExporter {
+    private[this] val singleThread: Boolean = settings.YaddBackendThreads.value > 1
+    private[this] val builder: StringBuilder = new StringBuilder(64)
+
+    final def btypeToString(btype: BType): String =
+      if (singleThread){
+        builder.setLength(0)
+        appendBType(builder, btype)
+        builder.toString
+      } else {
+        val ephBuilder = new StringBuilder(64)
```

Comment:
you could try using a `ThreadLocal` here to regain the benefit in the parallel case

Reply:
Unfortunately, I have no experience using `ThreadLocal` variables. Could you suggest (or commit) the changes that would be needed?

Reply:
I have applied the `ThreadLocal`, as suggested.